odoo/addons/base/models/res_users.py

```diff
@@ -624,6 +624,84 @@ def _is_superuser(self):
     def get_company_currency_id(self):
         return self.env.user.company_id.currency_id.id
 
+    def _add_missing_default_values(self, vals):
+        # we need to override this function to keep to have the value of groups_id
+        # supersede the value of inverse computed fields at user creation (in models.py)
+        res = super(Users, self)._add_missing_default_values(vals)
+        # we remove all default for group fields if groups_id was given and that field wasn't explicitly given
+        new_vals = {key: value for (key, value) in res.items()
+                    if 'groups_id' not in vals or key in vals or (not is_boolean_group(key) or not is_selection_groups(key))}
+        return new_vals
+
+    def _get_group_selection(self, category_xml_id):
+        """Returns the ordered selection for the given ir.module.category xmlid"""
+        category = self.env.ref(category_xml_id, raise_if_not_found=False)
+        selections = []
+        if category:
+            groups = self.env['res.groups'].search([('category_id', '=', category.id)])
+            order = {group: len(group.trans_implied_ids & groups) for group in groups}
+            groups = groups.sorted(key=order.get)
+            selections = [(group.id, group.name) for group in groups]
+        return selections
+
+    @api.depends('groups_id')
+    def _compute_groups_id(self):
+        """Set the value for the group field (Selection of Boolean)
+           according to the user's groups, defined by the field 'groups_id'."""
+        computed_group_fields = [field for field in self._field_computed.keys() if field.compute == '_compute_groups_id']
+        all_groups = self.env['res.groups'].search([])
+        for user in self:
+            for field in computed_group_fields:
+                # Checkbox group
+                if getattr(field, 'group_xml_id', None):
```

Comment:
I would separate fields-based computation from user update, to avoid prefetching issue leading to recomputing groups data in each loop iteration.